### PR TITLE
feat(gamut-styles): Added TypeScript .d.ts generation to gamut-styles

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Added TypeScript .d.ts generation to gamut-styles

<details><summary>The generated `.d.ts` uses `namespace`s for some reason. </summary>

```ts
export namespace colors {
    export const blue: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
        1000: string;
        1100: string;
    };
    export const pink: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const purple: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const red: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const green: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const orange: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const yellow: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const gray: {
        100: string;
        200: string;
        300: string;
        400: string;
        500: string;
        600: string;
        700: string;
        800: string;
        900: string;
    };
    export const black: string;
    export const white: string;
    export const beige: string;
    export const royalBlue: string;
}
export namespace brandColors {
    const red_1: string;
    export { red_1 as red };
    const orange_1: string;
    export { orange_1 as orange };
    const yellow_1: string;
    export { yellow_1 as yellow };
    const purple_1: string;
    export { purple_1 as purple };
    const pink_1: string;
    export { pink_1 as pink };
    export const mint: string;
    const beige_1: string;
    export { beige_1 as beige };
    export const darkBlue: string;
}
export namespace effectColors {
    export const slightShadow: string;
}
export namespace deprecatedColors {
    export namespace portal {
        const blue_1: string;
        export { blue_1 as blue };
        export const darkblue: string;
        export const midnightblue: string;
        const mint_1: string;
        export { mint_1 as mint };
    }
    export namespace swatches {
        export const ccblue: {
            500: string;
            700: string;
            900: string;
        };
    }
}
export namespace editorColors {
    const blue_2: string;
    export { blue_2 as blue };
    export const deepPurple: string;
    const gray_1: string;
    export { gray_1 as gray };
    const green_1: string;
    export { green_1 as green };
    const orange_2: string;
    export { orange_2 as orange };
    const purple_2: string;
    export { purple_2 as purple };
    const red_2: string;
    export { red_2 as red };
    const yellow_2: string;
    export { yellow_2 as yellow };
}
export namespace deprecatedGamutColors {
    export namespace base {
        const purple_3: string;
        export { purple_3 as purple };
        const royalBlue_1: string;
        export { royalBlue_1 as royalBlue };
        const red_3: string;
        export { red_3 as red };
        const mint_2: string;
        export { mint_2 as mint };
        const yellow_3: string;
        export { yellow_3 as yellow };
    }
    export namespace swatches_1 {
        const purple_4: {
            100: string;
            200: string;
            300: string;
            400: string;
            500: string;
            600: string;
            700: string;
            800: string;
            900: string;
        };
        export { purple_4 as purple };
        const royalBlue_2: {
            500: string;
            600: string;
            700: string;
            800: string;
        };
        export { royalBlue_2 as royalBlue };
        const mint_3: {
            100: string;
            200: string;
            300: string;
            400: string;
            500: string;
            600: string;
            700: string;
            800: string;
            900: string;
        };
        export { mint_3 as mint };
    }
    export { swatches_1 as swatches };
}
export namespace grid {
    export const cols: number;
    export const gutterWidth: string;
    export const outerMargin: string;
    export const maxContentWidth: string;
    export const xsMin: string;
    export const smMin: string;
    export const mdMin: string;
    export const lgMin: string;
    export const xlMin: string;
}
export namespace legacyBreakpoints {
    export const phone: string;
    export const tablet: string;
    export const desktop: string;
}
export namespace breakpoints {
    export const xs: string;
    export const sm: string;
    export const md: string;
    export const lg: string;
    export const xl: string;
}
declare namespace _default {
    export { colors };
    export { brandColors };
    export { deprecatedColors };
    export { deprecatedGamutColors };
    export { editorColors };
    export { effectColors };
    export { grid };
    export { breakpoints };
    export { legacyBreakpoints };
}
export default _default;
```

</details>

### CODE REVIEW

File: packages/gamut-styles/package.json [hunk 1]
```diff
@@ -5,15 +5,12 @@
   "author": "Jake Hiller <jake@codecademy.com>",
   "main": "utils.scss",
   "scripts": {
-    "develop-core": "npx node-sass core/index.scss demo-styles/index.css; npx http-server"
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/package.json:8]: @jakemhiller we can get rid of this `develop-core` & dependencies, right?

Comment (jakemhiller) [reply packages/gamut-styles/package.json:8]: yeah, definitely

**Comment by jakemhiller** [comment]
yesss

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb07c2698c8bafea620cb3d--codecademy-styleguide.netlify.app>https://5eb07c2698c8bafea620cb3d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb07c2698c8bafea620cb3d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.5.3-alpha.a00469.0<br/>@codecademy/gamut-styles@3.1.2-alpha.a00469.0<br/>@codecademy/gamut@7.4.6-alpha.a00469.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb07cd1cfa81402035fd783--codecademy-styleguide.netlify.app>https://5eb07cd1cfa81402035fd783--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb07cd1cfa81402035fd783>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.5.3-alpha.5a2fe2.0<br/>@codecademy/gamut-styles@3.1.2-alpha.5a2fe2.0<br/>@codecademy/gamut@7.4.6-alpha.5a2fe2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb082276ef43801f5f6b08a--codecademy-styleguide.netlify.app>https://5eb082276ef43801f5f6b08a--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb082276ef43801f5f6b08a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.5.3-alpha.52ad8f.0<br/>@codecademy/gamut-styles@3.1.2-alpha.52ad8f.0<br/>@codecademy/gamut@7.4.6-alpha.52ad8f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb08dc46ef438c953f6acc9--codecademy-styleguide.netlify.app>https://5eb08dc46ef438c953f6acc9--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb08dc46ef438c953f6acc9>Deploy Logs</a></p>

**Comment by hmafzal** [comment]
Ugh yes

**Pull request merged**